streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamTransformValues.java

```diff
@@ -168,16 +168,12 @@ public long timestamp() {
 
         @Override
         public void process(K key, V value) {
-            context.forward(key, valueTransformer.transform(value));
+            context.forward(key, valueTransformer.transform(key, value));
         }
 
         @SuppressWarnings("deprecation")
         @Override
-        public void punctuate(long timestamp) {
-            if (valueTransformer.punctuate(timestamp) != null) {
-                throw new StreamsException("ValueTransformer#punctuate must return null.");
-            }
-        }
+        public void punctuate(long timestamp) {}
```

Comment:
I think it is probably worth adding as even if it is deprecated it is still supported

Reply:
+1, agree with @mjsax and @dguy

Reply:
+1 agree as well

Reply:
@mjsax I have some doubts about the solution you proposed. Here, independent of user's using  `ValueTransformerWithKeySupplier` or `ValueTransformerSupplier`, `KStreamTransformValues` class always gets `ValueTransformerWithKeySupplier`. The reason is that we always wrap `ValueTransformerSupplier` into `ValueTransformerWithKeySupplier` and pass and the latter one to `KStreamTransformValues`. Moreover, `ValueTransformerWithKeySupplier` and `ValueTransformerSupplier` are separate interfaces, they don't extend from each other. 
So, I think the most feasible solution for that is to include deprecated method (`punctuate()`) in `ValueTransformerWithKeySupplier`.

Reply:
Because `ValueTransformerWithKeySupplier` is a public interface, we should try to find a solution that does not add a deprecated method to this new interface. If my proposal doesn't work, I am sure there is another solution (using sub-classing etc) to do some internal re-directs to make it work.